File: WebsiteSettings.cs
using System.IO;
using System.Text.Json;
using Saber.Core;
using Saber.Vendor;

namespace Saber.Vendors.StressTest
{
    public class WebsiteSettings : IVendorWebsiteSettings
    {
        public string Name { get; set; } = "Stress Testing";
        public string Render(IRequest request)
        {
            if (!request.CheckSecurity("stress-test")) { return ""; }
            var view = new View(App.MapPath("/Vendors/StressTest/websitesettings.html"));
            var file = App.MapPath("/Vendors/StressTest/config.json");
            var prepared = false;
            if (File.Exists(file))
            {
                var config = JsonSerializer.Deserialize<Config>(File.ReadAllText(file));
                prepared = config.Prepared;
            }
            if (!prepared)
            {
                view.Show("prepare");
            }
            else
            {
                view.Show("run");
            }
            request.AddScript("/editor/vendors/stresstest/websitesettings.js");
            return view.Render();
        }
    }
}

File: websitesettings.js
(function () {
    S('.prepare-stress-test button').on('click', () => {
        S.ajax.post('StressTest/Prepare', {}, () => {
            S('.prepare-stress-test').addClass('hide');
            S('.run-stress-test').removeClass('hide');
            S('.delete-stress-test').removeClass('hide');
        })
    });

    S('.delete-stress-test button').on('click', () => {
        S.ajax.post('StressTest/Delete', {}, () => {
            S('.prepare-stress-test').removeClass('hide');
            S('.run-stress-test').addClass('hide');
            S('.delete-stress-test').addClass('hide');
        })
    });
})();